pkg/controllers/jobset_controller.go

```diff
@@ -307,63 +307,112 @@ func (r *JobSetReconciler) calculateReplicatedJobStatuses(ctx context.Context, j
 
 func (r *JobSetReconciler) suspendJobSet(ctx context.Context, js *jobset.JobSet, ownedJobs *childJobs) error {
 	for _, job := range ownedJobs.active {
-		if !ptr.Deref(job.Spec.Suspend, false) {
+		if !jobSuspended(job) {
 			job.Spec.Suspend = ptr.To(true)
 			if err := r.Update(ctx, job); err != nil {
 				return err
 			}
 		}
 	}
-	return r.ensureCondition(ctx, js, corev1.EventTypeNormal, metav1.Condition{
-		Type:               string(jobset.JobSetSuspended),
-		Status:             metav1.ConditionStatus(corev1.ConditionTrue),
-		LastTransitionTime: metav1.Now(),
-		Reason:             "SuspendedJobs",
-		Message:            "jobset is suspended",
+	return r.ensureCondition(ctx, ensureConditionOpts{
+		jobset:           js,
+		eventType:        corev1.EventTypeNormal,
+		forceFalseUpdate: false,
```

Comment:
This is false by default, so no need to set it. Here and many other places in the file